src/core/coordinators/streams/stream.ts

```diff
@@ -0,0 +1,382 @@
+    const oldEnabledRoutes = this.streamConfig.enabledRoutes;
+    const oldRespectPageVisibility = this.config.respectPageVisibility;
+
+    // Update base config (handles intervalMs restart logic)
+    super.configure(config);
+
+    // Update stream-specific config
+    if (config.enabledRoutes) {
+      this.streamConfig.enabledRoutes = config.enabledRoutes;
+    }
+    if (config.fetchLimit !== undefined) {
+      this.streamConfig.fetchLimit = config.fetchLimit;
+    }
+
+    // Re-evaluate polling state if config changes affect polling conditions
+    // This ensures changes take effect immediately without manual stop/start
+    const enabledRoutesChanged = config.enabledRoutes && oldEnabledRoutes !== this.streamConfig.enabledRoutes;
+    const visibilityConfigChanged =
+      config.respectPageVisibility !== undefined && oldRespectPageVisibility !== this.config.respectPageVisibility;
+
+    if (enabledRoutesChanged || visibilityConfigChanged) {
+      void this.evaluateAndStartPolling();
+    }
+  }
+
+  // ============================================================================
+  // Private API
+  // ============================================================================
+
+  /**
+   * Resolve the stream ID based on current route
+   * Returns null if stream ID cannot be determined
+   * Resets streamHead when stream ID changes to avoid using stale head values
+   */
+  private resolveStreamId() {
+    const previousStreamId = this.streamState.currentStreamId;
+
+    // Home route: build from home store state
+    if (this.state.currentRoute === APP_ROUTES.HOME) {
+      const { sort, reach, content } = Core.useHomeStore.getState();
+      this.streamState.currentStreamId = Core.getStreamId(sort, reach, content);
+      Logger.debug(`Built ${APP_ROUTES.HOME} streamId`, { streamId: this.streamState.currentStreamId });
+    }
+    // Post route: extract from URL and build reply stream ID
+    else if (this.state.currentRoute.startsWith(POST_ROUTES.POST)) {
+      this.buildPostReplyStreamId();
+      Logger.debug(`Built ${POST_ROUTES.POST} streamId`, { streamId: this.streamState.currentStreamId });
+    } else {
+      this.streamState.currentStreamId = null;
+    }
+
+    // Reset streamHead if stream ID changed
+    if (previousStreamId !== this.streamState.currentStreamId) {
+      this.streamState.streamHead = Core.SKIP_FETCH_NEW_POSTS;
+      if (previousStreamId !== null || this.streamState.currentStreamId !== null) {
+        Logger.debug('Stream ID changed, reset stream head', {
+          previousStreamId,
+          newStreamId: this.streamState.currentStreamId,
+        });
+      }
+    }
+  }
+
+  /**
+   * Resolve the stream head based on current stream ID
+   */
+  private async resolveStreamHead(currentStreamId: Core.PostStreamId): Promise<boolean> {
+    const streamHead = await Core.StreamPostsController.getStreamHead({ streamId: currentStreamId });
```

Comment:
No try-catch around `getStreamHead`. If it throws, the error bubbles up and may crash the polling interval.

```typescript
private async resolveStreamHead(currentStreamId: Core.PostStreamId): Promise<boolean> {
  try {
    const streamHead = await Core.StreamPostsController.getStreamHead({ streamId: currentStreamId });
    // ...
  } catch (error) {
    Logger.error('Failed to resolve stream head', { streamId: currentStreamId, error });
    return false;
  }
}
```